### ISSUE TITLE
Allow update methods to change the schema using arguments.

### ISSUE THREAD

**Issue by fredysomy** [fredysomy/pysonDB#48]
### Discussed in https://github.com/fredysomy/pysonDB/discussions/43

<div type='discussions-op-text'>

<sup>Originally posted by **fredysomy** September 30, 2021</sup>
**How would this feature be useful?**

In case the developer need to add a extra key (in terms of dictionaries) , the current methods throw error if the schema is violated.

**Describe the solution you'd like**
Allow update method to modify the schemma . (Should be optional (using args))

</div>

**Comment by pranavbaburaj** [comment]
Hmm. maybe we can let the user change the keys with a default value which can be assigned to the existing fields.

**Comment by fredysomy** [comment]
Yes exactly , or lets say 
* The user needs to have a default value there (we let the user do that.)
* And if the user didnt give any default, we can simply assign it null.

**Comment by fredysomy** [comment]
I had a thought about this and this is how my idea looks like.
```
Db.update({"non-exisstent-key":"default/user choice"},addKeyIfDoesntExist=true)
```

We can check for that arg there and if it is there we can add the key and default value to the fields, else we raise a schemma error.

**Comment by fredysomy** [comment]
What do you think @Adwaith-Rajesh @DanBrown47 . Or you guys have some other workaround.

**Comment by pranavbaburaj** [comment]
> I had a thought about this and this is how my idea looks like.
> 
> ```
> Db.update({"non-exisstent-key":"default/user choice"},addKeyIfDoesntExist=true)
> ```
> 
> We can check for that arg there and if it is there we can add the key and default value to the fields, else we raise a schemma error.

In that example, we will have to set the field value to null. But, I think we can also add a method to create a new field with a default value.

Something like this?
```py
db.add_key(key:str, default:Any=None)
```

**Comment by fredysomy** [comment]
Yes, but think of this, this schemma changing scenario is happening in the update method, so if we can fit in the schemma changing methods into the update method , it will be more usefull for users .

**Comment by pranavbaburaj** [comment]
Yeah. you are right. that would actually be easier.

**Comment by pranavbaburaj** [comment]
I have another suggestion. I have created a new issue(#53)

**Comment by Adwaith-Rajesh** [comment]
We can implement this.

**Comment by 3dw1np** [comment]
Any update on that ?

**Comment by DanBrown47** [comment]
This is a cool feature ! I think it will be awesome to move forward with this
